pkgs/applications/misc/collision/default.nix

```diff
@@ -0,0 +1,159 @@
+{ lib
+, fetchFromGitHub
+, fetchgit
+, crystal
+, libadwaita
+, gtk4
+, gtk3
+, glib
+, gobject-introspection
+, openssl
+, libxml2
+, pkg-config
+, wrapGAppsHook4
+}:
+
+crystal.buildCrystalPackage rec {
+  pname = "Collision";
+  version = "3.5.0";
+
+  # project source
+  src = fetchFromGitHub {
+    owner = "GeopJr";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    name = pname;
+    hash = "sha256-YNMtiMSzDqBlJJTUntRtL6oXg+IuyAobQ4FYcwOdOas=";
+    };
+
+  # GTK4/libadwaita binding generator for Crystal
+  gi_crystal = fetchFromGitHub {
+      owner = "hugopl";
+      repo = "gi-crystal";
+      rev = "refs/tags/v0.16.0";
+      name = "gi-crystal";
+      hash = "sha256-ij4U8BoSNHpkh5SmvoH5anGB/ZdvmC5zDwJCusR/s/c=";
+    };
+
+  # project dependencies
+  dependencies = [
+    (fetchgit {
+      url = "https://github.com/GeopJr/libadwaita.cr";
+      name = "libadwaita";
+      hash = "sha256-3F1UrApfTQn4kl64PtuQJW/V5XzfToYPqRqT60E2XyY=";
+    })
+    (fetchgit {
+      url = "https://github.com/hugopl/harfbuzz.cr";
+      name = "harfbuzz";
+      hash = "sha256-23CNz9iS9sTw4IYngWq9arH7Kdgqpo/XiDOKom1Z7xA=";
+    })
```

Comment:
Please prefer `fetchFromGitHub` here, with a proper tag or commit.

If there are no tags in the repo, please ask upstream to create them.

Use a commit SHA1 until that has been solved. Make sure to link the issue you created in a comment.

Reply:
Fetchgit removed in https://github.com/NixOS/nixpkgs/pull/242681/commits/1de87e0e063a584308d57a83df37c752e7968fcb